src/algebra/big_operators/basic.lean

```diff
@@ -802,21 +802,25 @@ lemma mul_prod_diff_singleton [decidable_eq α] {s : finset α} {i : α} (h : i
   (f : α → β) : f i * (∏ x in s \ {i}, f x) = ∏ x in s, f x :=
 by { convert s.prod_inter_mul_prod_diff {i} f, simp [h] }
 
+/-- Partition a product by a setoid. -/
+@[to_additive "Partition a sum by a setoid"]
```

Comment:
@robertylewis, is this better?
```suggestion
/-- Partition a product into a product of products, each equivalent under a setoid. -/
@[to_additive "Partition a sum into a sum of sums, each equivalent under a setoid"]
```

Reply:
```suggestion
/-- A product can be partitioned into a product of products, each equivalent under a setoid. -/
@[to_additive "A sum can be partitioned into a sum of sums, each equivalent under a setoid."]
```
? It's stating a fact, not defining an operation. At least it matches the grammar of the next doc string better.

Reply:
Ah, I see your point - I was writing the docstring for users of `rw prod_partition`, but that's obviously not the right way to document things.